### PR TITLE
prettier-comments

### PR DESCRIPTION
**Pull request by mattiaerre**
# Description

this PR is to show how to integrate and use `prettier-comments`, have a look at [https://github.com/prettier-solidity/prettier-comments](https://github.com/prettier-solidity/prettier-comments) for more details

- relates to #70

### CODE REVIEW

**Comment by j-f1** [comment]
cc @prettier/core do you want to extract these to a package under @prettier?

File: tests/Etc/Etc.sol [hunk 1]
```diff
@@ -32,4 +32,11 @@ contract Contract {
     uint sum;
     for ( i = 0; i < 10; i++ ) sum   +=  i;
   }
+
+  function fun(uint256 a) returns (uint) {
+    if (something) foo();
+    // comment
```
Comment (fvictorio) [inline tests/Etc/Etc.sol]: Shouldn't there be an updated snapshot for this too?

File: src/index.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const { handleComments } = require('prettier-comments'); // eslint-disable-line import/no-extraneous-dependencies
```
Comment (fvictorio) [inline src/index.js]: Any reason for the extraneous dependency?

Comment (fvictorio) [reply src/index.js]: Oh, nevermind, I see that you are using it linked for now.

File: tests/Etc/Etc.sol (hunk 1 above)
Comment (mattiaerre) [reply tests/Etc/Etc.sol]: nope as it throws an error now as per:

https://github.com/prettier-solidity/prettier-comments#how-to-use-it-w-prettier-plugin-solidity

basically the `comments.js` I've extracted from `prettier` knows how to scan their AST but not the one we are using (this is my assumption)

**Comment by mattiaerre** [comment]
No joy from the prettier team so I've decided to move the prettier-comments source into this project as I didn't want (not sure if I am allowed) to publish a separate module; I've managed to keep everything separate so that if there will be a module, we would just need to change the import; let me know what you think so that I can publish this version that fixes the comments issue.

// cc @j-f1 @fvictorio

File: tests/Etc/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -35,6 +35,13 @@ contract Contract {
     uint sum;
     for ( i = 0; i < 10; i++ ) sum   +=  i;
   }
+
+  function fun(uint256 a) returns (uint) {
+    if (something) foo();
+    // comment
```
Comment (mattiaerre) [inline tests/Etc/__snapshots__/jsfmt.spec.js.snap]: YAY!

**Pull request merged**